### PR TITLE
net-dialup/accel-ppp: revbumping and fix building on musl

### PR DESCRIPTION
**Pull request by listout**
Patches taken from alpine linux to fix build on musl. Only major change
is making accel-ppp on libucontext.

Closes: https://bugs.gentoo.org/828906

Signed-off-by: brahmajit das <brahmajit.xyz@gmail.com>

### CODE REVIEW

File: net-dialup/accel-ppp/accel-ppp-1.12.0_p20210430.ebuild [hunk 1]
```diff
@@ -23,7 +23,8 @@ RDEPEND="!libtomcrypt? ( dev-libs/openssl:0= )
 	snmp? ( net-analyzer/net-snmp )
 	dev-libs/libpcre"
 DEPEND="${RDEPEND}
-	valgrind? ( dev-util/valgrind )"
+	valgrind? ( dev-util/valgrind )
+	sys-libs/libucontext"
```
Comment (thesamesam) [inline net-dialup/accel-ppp/accel-ppp-1.12.0_p20210430.ebuild:27]: libucontext needs to be in RDEPEND too, but this should be elibc_musl? ( ... )

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: net-dialup/accel-ppp/accel-ppp-1.12.0_p20210430.ebuild (hunk 1 above)
Comment (listout) [reply net-dialup/accel-ppp/accel-ppp-1.12.0_p20210430.ebuild]: Added

File: net-dialup/accel-ppp/accel-ppp-1.12.0_p20210430-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+RDEPEND="!libtomcrypt? ( dev-libs/openssl:0= )
+	libtomcrypt? ( dev-libs/libtomcrypt:0= )
+	lua? ( ${LUA_DEPS} )
+	postgres? ( dev-db/postgresql:* )
+	snmp? ( net-analyzer/net-snmp )
+	dev-libs/libpcre
+	elibc_musl? ( sys-libs/libucontext )"
+DEPEND="${RDEPEND}
+	valgrind? ( dev-util/valgrind )
+	elibc_musl? ( sys-libs/libucontext )"
```
Comment (thesamesam) [inline net-dialup/accel-ppp/accel-ppp-1.12.0_p20210430-r1.ebuild:28]: This is still in DEPEND only, my comment earlier said about moving it into RDEPEND as it is linked against it

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#
+# Closes: https://bugs.gentoo.org/828906
+--- a/accel-pppd/CMakeLists.txt
++++ b/accel-pppd/CMakeLists.txt
+@@ -113,7 +113,7 @@ ADD_EXECUTABLE(accel-pppd
+ 	main.c
+ )
+
+-TARGET_LINK_LIBRARIES(accel-pppd triton rt pthread ${crypto_lib} pcre)
++TARGET_LINK_LIBRARIES(accel-pppd triton rt pthread ${crypto_lib} pcre ucontext)
```
Comment (thesamesam) [inline net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch:14]: You sure this works on glibc?

Comment (listout) [reply net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch:14]: Hhh

File: net-dialup/accel-ppp/accel-ppp-1.12.0_p20210430-r1.ebuild (hunk 2 above)
Comment (listout) [reply net-dialup/accel-ppp/accel-ppp-1.12.0_p20210430-r1.ebuild:28]: Okay. I'll do that

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch (hunk 3 above)
Comment (thesamesam) [reply net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch:14]: If it builds, it should be OK.

Comment (listout) [reply net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch]: The ucontext inclusion was conditional and I made the mistake of testing on Glibc while libucontext installed as a result couldn't catch that bug on glibc. Should be fixed with latest push. libucontext inclusion is now conditional.

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+@@ -113,7 +113,21 @@ ADD_EXECUTABLE(accel-pppd
+ 	main.c
+ )
+
++option(USE_LIBUCONTEXT "Use libucontext on non glibc systems" OFF)
++IF(USE_LIBUCONTEXT)
++	find_library(LIBUCONTEXT_LIBRARY ucontext)
++
++	IF(NOT LIBUCONTEXT_LIBRARY)
++		message(FATAL_ERROR "${Esc}[31mRequired libucontext not found.\n Install ucontext and run cmake again${Esc}[m")
```
Comment (thesamesam) [inline net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch:18]: It's a nice touch, but i wouldn't bother with the colours.

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# Use ucontext, this is specific to musl libc
+#
+# Alpine patch:
```
Comment (thesamesam) [inline net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch:3]: I'd say "adapted from"

File: net-dialup/accel-ppp/accel-ppp-1.12.0_p20210430-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+		-DBUILD_VLAN_MON_DRIVER="$(usex ipoe)"
+		-DCRYPTO="$(usex libtomcrypt TOMCRYPT OPENSSL)"
+		-DLOG_PGSQL="$(usex postgres)"
+		-DLUA="$(usex lua TRUE FALSE)"
+		-DMEMDEBUG="$(usex debug)"
+		-DNETSNMP="$(usex snmp)"
+		-DRADIUS="$(usex radius)"
+		-DSHAPER="$(usex shaper)"
+		$(use debug && echo "-DVALGRIND=$(usex valgrind)")
+		$(use elibc_musl && echo "-DUSE_LIBUCONTEXT=ON")
```
Comment (thesamesam) [inline net-dialup/accel-ppp/accel-ppp-1.12.0_p20210430-r1.ebuild:91]: Yours is fine, but let's do:
```suggestion
		-DUSE_LIBUCONTEXT="$(usex elibc_musl)"
```

Comment (listout) [reply net-dialup/accel-ppp/accel-ppp-1.12.0_p20210430-r1.ebuild:91]: Sure

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch (hunk 4 above)
Comment (listout) [reply net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch:18]: I was thinking of upstream ing the patch hence stuck with what they were using

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch (hunk 5 above)
Comment (listout) [reply net-dialup/accel-ppp/files/accel-ppp-1.12.0-ucontext.patch:3]: Sure, I'll change it

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-printf.patch [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# No need to include printf.h
+#
+# Adapted from Alpine linux patch:
```
Comment (thesamesam) [inline net-dialup/accel-ppp/files/accel-ppp-1.12.0-printf.patch:3]: Was this one really adapted or taken literally?

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-if_arp.patch [hunk 8]
```diff
@@ -0,0 +1,30 @@
+# Dont use if_arp on musl libc. Causes redefinition error
+#
+# Adapted from Alpine linux patch:
```
Comment (thesamesam) [inline net-dialup/accel-ppp/files/accel-ppp-1.12.0-if_arp.patch:3]: Was this one really adapted or taken literally?

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-tsearch.patch [hunk 9]
```diff
@@ -0,0 +1,20 @@
+# Don't type cast l2tp_tunnel_free_sessions on musl libc
+#
+# Adapted from Alpine linux patch:
```
Comment (thesamesam) [inline net-dialup/accel-ppp/files/accel-ppp-1.12.0-tsearch.patch:3]: Was this one really adapted or taken literally?

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-if_arp.patch (hunk 8 above)
Comment (listout) [reply net-dialup/accel-ppp/files/accel-ppp-1.12.0-if_arp.patch:3]: What do you think should it be @thesamesam

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-printf.patch (hunk 7 above)
Comment (listout) [reply net-dialup/accel-ppp/files/accel-ppp-1.12.0-printf.patch:3]: This is taken literally

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-if_arp.patch (hunk 8 above)
Comment (thesamesam) [reply net-dialup/accel-ppp/files/accel-ppp-1.12.0-if_arp.patch]: It's fine if you really did adapt it :)

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-printf.patch (hunk 7 above)
Comment (thesamesam) [reply net-dialup/accel-ppp/files/accel-ppp-1.12.0-printf.patch]: Then don't say "adapted" :)

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-if_arp.patch (hunk 8 above)
Comment (listout) [reply net-dialup/accel-ppp/files/accel-ppp-1.12.0-if_arp.patch]: Maybe modified would be a better word

File: net-dialup/accel-ppp/files/accel-ppp-1.12.0-tsearch.patch [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+# https://git.alpinelinux.org/aports/tree/community/accel-ppp/0003-tsearch.patch
+#
+# Closes: https://bugs.gentoo.org/828906
+--- a/accel-pppd/ctrl/l2tp/l2tp.c
++++ b/accel-pppd/ctrl/l2tp/l2tp.c
+@@ -852,7 +852,11 @@ static void l2tp_tunnel_free_sessions(struct l2tp_conn_t *conn)
+ 	void *sessions = conn->sessions;
+
+ 	conn->sessions = NULL;
++#if defined(__GLIBC__)
```
Comment (thesamesam) [inline net-dialup/accel-ppp/files/accel-ppp-1.12.0-tsearch.patch:13]: Is the cast really needed on glibc? This feels like the kind of thing we should be able to do correctly unconditionally.

Comment (listout) [reply net-dialup/accel-ppp/files/accel-ppp-1.12.0-tsearch.patch:13]: Yes, it's needed in glibc. Else it throws error

Comment (listout) [reply net-dialup/accel-ppp/files/accel-ppp-1.12.0-tsearch.patch:13]: Huh, @thesamesam. You're right, it doesn't give any error without the cast on glibc. I thought I previously faced a build error with the cast on glibc.

**Pull request closed without merging**